frontend/packages/console-app/src/hooks/useNotificationPoller.ts

```diff
@@ -0,0 +1,123 @@
+
+  // Update alert count.
+  const alertCount = alerts?.length ?? 0;
+  React.useEffect(() => {
+    dispatch(setAlertCount(alertCount));
+  }, [alertCount, dispatch]);
+
+  React.useEffect(() => {
+    const pollerTimeouts = {};
+    const pollers = {};
+
+    const poll: NotificationPoll = (url, key: 'notificationAlerts' | 'silences', dataHandler) => {
+      dispatch(alertingLoading(key));
+      const notificationPoller = (): void => {
+        coFetchJSON(url)
+          .then((response) => dataHandler(response))
+          .then((data) => {
+            dispatch(alertingLoaded(key, data));
+            pollerTimeouts[key] = setTimeout(notificationPoller, 15 * 1000);
+          })
+          .catch((e) => {
+            dispatch(alertingErrored(key, e));
+
+            // If the API returned an error, poll less frequently to avoid excessive calls. For
+            // example, if the user doesn't have permission to access the API, polling will probably
+            // continue to fail, but it is possible that permissions will be granted so we don't
+            // stop completely.
+            pollerTimeouts[key] = setTimeout(notificationPoller, 60 * 1000);
+          });
+      };
+      pollers[key] = notificationPoller;
+      notificationPoller();
+    };
+
+    const { alertManagerBaseURL, prometheusBaseURL } = window.SERVER_FLAGS;
+
+    if (prometheusBaseURL) {
+      poll(
+        `${prometheusBaseURL}/${PrometheusEndpoint.RULES}`,
+        'notificationAlerts',
+        (alertsResults: PrometheusRulesResponse): Alert[] =>
+          alertsResults
+            ? getAlertsAndRules(alertsResults.data)
+                .alerts.filter(
+                  (a) =>
+                    a.state === 'firing' &&
+                    getAlertName(a) !== 'Watchdog' &&
+                    getAlertName(a) !== 'UpdateAvailable',
+                )
+                .sort((a, b) => +new Date(getAlertTime(b)) - +new Date(getAlertTime(a)))
+            : [],
+      );
+    } else {
+      dispatch(
+        alertingErrored('notificationAlerts', new Error(t('public~prometheusBaseURL not set'))),
+      );
+    }
+
+    if (alertManagerBaseURL) {
+      poll(`${alertManagerBaseURL}/api/v2/silences`, 'silences', (silences) => {
+        // Set a name field on the Silence to make things easier
+        _.each(silences, (s) => {
+          s.name = _.get(_.find(s.matchers, { name: 'alertname' }), 'value');
+          if (!s.name) {
+            // No alertname, so fall back to displaying the other matchers
+            s.name = s.matchers
+              .map(
+                (m) => `${m.name}${silenceMatcherEqualitySymbol(m.isEqual, m.isRegex)}${m.value}`,
+              )
+              .join(', ');
+          }
+        });
+        return silences;
+      });
+    } else {
+      dispatch(alertingErrored('silences', new Error(t('public~alertManagerBaseURL not set'))));
+    }
+
+    return () => {
+      _.each(pollerTimeouts, clearTimeout);
+    };
```

Comment:
Had to rebase due to Jacksons [PR](https://github.com/openshift/console/pull/14864), this return syntax was the only conflicting change - no other changes were brought up. 
before it looked like this
```suggestion
    return () => _.each(pollerTimeouts, clearTimeout);
;
```